models/_generate_layers.py

```diff
@@ -496,6 +497,38 @@ def collect_layers():
   return ls
 
 
+def get_super_call_params(layer):
+  """
+  Inspects the given layer, extracts unwanted parameters and then returns a string that is ready to be used in the super
+  call.
+  :param layer: Layer to be inspected
+  :return: string with super call parameters
+  """
+  # get code as string list
+  code = inspect.getsource(layer.__init__).splitlines()
+
+  # get list of lines starting with super and take the first one
+  lines = [line.strip() for line in code if line.strip().startswith("super")]
+  if not lines:
+    return ''
+
+  # reformat the super call to extract what we need
+  super_call = lines[0]
+  call = super_call.split(".")[1]
+  call_pruned = call[9:-1]
+
+  # remove excluded params
+  split = call_pruned.split(",")
+  tup_ls = list(map(lambda x: x.split("="), split))
+  for idx, param in enumerate(tup_ls):
```

Comment:
You splitted the `param` before by `=`. Then use it more explicitly here, which makes it easier. E.g.:
```suggestion
  for idx, (key, value) in enumerate(tup_ls):
```